transform-plugins/src/main/java/io/cdap/plugin/UnionSplitter.java

```diff
@@ -104,28 +103,36 @@ public void transform(StructuredRecord record, MultiOutputEmitter<StructuredReco
       valSchema = Schema.of(Schema.Type.DOUBLE);
     } else if (val instanceof String) {
       valSchema = Schema.of(Schema.Type.STRING);
-    } else if (val instanceof StructuredRecord) {
+    }else if (val instanceof Time) {
```

Comment:
missing a space after `}`

Reply:
Checking the instance of the `val` won't work. This is because calling `Object val = record.get(conf.unionField);` would return the numeric value, instead of the logical type class. You should use the schema type of the union field in the incoming `record` as the schema of the output value.